docs/mp/integrations/neo4j.adoc

```diff
@@ -0,0 +1,268 @@
+                    + "match (m) <- [:DIRECTED] - (d:Person) "
+                    + "match (m) <- [r:ACTED_IN] - (a:Person) "
+                    + "return m, collect(d) as directors, collect({name:a.name, roles: r.roles}) as actors";
+
+            return session.readTransaction(tx -> tx.run(query).list(r -> {
+                var movieNode = r.get("m").asNode();
+
+                var directors = r.get("directors").asList(v -> {
+                    var personNode = v.asNode();
+                    return new Person(personNode.get("born").asInt(), personNode.get("name").asString());
+                });
+
+                var actors = r.get("actors").asList(v -> {
+                    return new Actor(v.get("name").asString(), v.get("roles").asList(Value::asString));
+                });
+
+                var m = new Movie(movieNode.get("title").asString(), movieNode.get("tagline").asString());
+                m.setReleased(movieNode.get("released").asInt());
+                m.setDirectorss(directors);
+                m.setActors(actors);
+                return m;
+            }));
+        }
+    }
+}
+
+----
+<1> `Neo4j` Driver constructor injection
+<2> Use `Neo4j` Driver to extract all Movies
+
+Movies now ca be returned as JSON objects:
+
+[source,java]
+----
+@GET
+@Produces(MediaType.APPLICATION_JSON)
+public List<Movie> getAllMovies() {
+    return movieRepository.findAll();
+}
+----
+
+Now build and run with JDK17+
+[source,bash]
+----
+mvn package
+java -jar target/helidon-examples-integration-neo4j-mp.jar
+----
+
+Exercise the application:
+
+[source,bash]
+----
+curl -X GET http://localhost:8080/movies
+{. . .}
+
+# Try health and metrics
+
+curl -s -X GET http://localhost:8080/health
+{"outcome":"UP",...
+. . .
+
+# Prometheus Format
+curl -s -X GET http://localhost:8080/metrics
+# TYPE base:gc_g1_young_generation_count gauge
+. . .
+
+# JSON Format
+curl -H 'Accept: application/json' -X GET http://localhost:8080/metrics
+{"base":...
+. . .
+
+----
+
+Full example code is available in link:https://github.com/oracle/helidon/tree/master/examples/integrations/neo4j/neo4j-mp[Helidon GitHub Repository].
+
+
+== Additional Information
+
+=== Neo4j Metrics propagation
+
+Neo4j metrics can be propagated to user as `MicroProfile` metrics. This is implemented in a separate Maven module. Just add
+
+[source,xml]
+----
+<dependency>
+   <groupId>io.helidon.integrations.neo4j</groupId>
+   <artifactId>helidon-integrations-neo4j-metrics</artifactId>
+</dependency>
+----
+
+To enable metrics in Neo4j, add the following property to `microprofile-config.properties`:
+
+[source,properties]
+----
+neo4j.pool.metricsEnabled=true
+----
+
+By applying these two actions, Neo4j metrics will be automatically added to the output of the `/metrics` endpoint.
+
+=== Neo4j Health Checks
+
+If your application is highly dependent on Neo4j database, health and liveness checks are essential for the correct work of this application.
```

Comment:
Maybe "...are essential for this application to work correctly."

Reply:
fixed